src/log/log.c

```diff
@@ -1744,6 +1745,22 @@ __wt_log_scan(WT_SESSION_IMPL *session, WT_LSN *lsnp, uint32_t flags,
 		    &rd_lsn, WT_LOG_FILENAME, 0));
 
 err:	WT_STAT_FAST_CONN_INCR(session, log_scans);
+	/*
+	 * If the first attempt to read a log record results in
+	 * an error recovery is likely going to fail.  Try to provide
+	 * a helpful failure message.
+	 */
+	if (ret != 0 && F_ISSET(log, WT_LOG_NOT_VERIFIED)) {
+		__wt_errx(session,
+		    "WiredTiger is unable to read the recovery log.");
+		__wt_errx(session, "This may be due to the log"
+		    " files being encrypted, being from an older"
+		    " version or due to corruption on disk");
+		__wt_errx(session, "You should confirm that you have"
+		    " opened the database with the correct options including"
+		    " all encryption and compression options");
+	}
+	F_CLR(log, WT_LOG_NOT_VERIFIED);
```

Comment:
Thanks, @sueloverso -- this lgtm.  It isn't an issue for now because `\_\_wt_log_scan` isn't in a performance path, but consider only clearing the flag if it is set, otherwise all callers will unconditionally update a shared cache line.